### PR TITLE
Support custom registries config

### PR DESCRIPTION
**Pull request by addyess**
Support setting up custom registries and authentication via a new option to configure the [`containerd-tempate.toml`](https://microk8s.io/docs/dockerhub-limits)

```bash
juju config microk8s custom_registries=\
  '[{"host": "docker.io", "url": "registry-1.docker.io", "username": "MYUSERNAME", "password": "MYPASSWORD"}]'
```

Without this, each unit will need to add the changes to the `containerd-template.toml` to add a custom registry or authentication for docker.io.  Use this to config to write those changes to containerd template for you.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -24,6 +24,51 @@ options:
...
 
+  custom_registries:
+    type: string
+    default: "[]"
+    description: |+
+      Registry endpoints and credentials. Setting this config allows containerd
+      to pull images from registries where auth is required.
+
+      The value for this config must be a JSON array of credential objects, like this:
+        e.g.: [{"host": "my.registry:port", "username": "user", "password": "pass"}]
```
Comment (stonepreston) [inline config.yaml:35]:
```suggestion
        e.g.: [{"url": "https://registry.example.com", "host": "my.registry:port", "username": "user", "password": "pass"}]
```

**Review by stonepreston — COMMENTED** [review]
small issue with required field missing in configuration comments, but this LGTM if its good with microk8s folks

File: tests/test_custom_registries.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+#         ]
+#     )
+
+#     with tempfile.TemporaryDirectory() as tmp_dir:
+#         with mock.patch("reactive.containerd.CONFIG_DIRECTORY", tmp_dir):
+#             containerd.config_changed()
+#         f_name = f"nvidia-{gpu}-{version}-config.toml"
+#         expected = pathlib.Path(__file__).parent / "test_custom_registries_render" / f_name
+#         target = pathlib.Path(tmp_dir) / "config.toml"
+#         assert list(target.open()) == list(expected.open())
```
Comment (neoaggelos) [inline tests/test_custom_registries.py:125]: Should this be removed?

**Review by neoaggelos — COMMENTED** [review]
Nice work! Looks good but need one extra comment to be addressed

File: tests/test_custom_registries.py (hunk 2 above)
Comment (addyess) [reply tests/test_custom_registries.py:125]: i.... i don't remember 😂 
I'll raise another commit here to figure out why i checked in a commented out test

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**